pagination/src/main/java/com/cerner/beadledom/pagination/OffsetPaginationModule.java

```diff
@@ -0,0 +1,141 @@
+package com.cerner.beadledom.pagination;
+
+import com.cerner.beadledom.pagination.parameters.LimitParameter;
+import com.cerner.beadledom.pagination.parameters.OffsetParameter;
+import com.google.inject.AbstractModule;
+import com.google.inject.Provides;
+import com.google.inject.name.Named;
+import javax.inject.Singleton;
+
+/**
+ * Module to provide the necessary configurations for using Offset Based Pagination.
+ *
+ * <p>Module provides the configurations needed to use offset based pagination. This includes
+ * the {@link OffsetPaginatedListLinksWriterInterceptor} as well as default values for the limit and
+ * offset fields and their respective field names. The default values for the limit, offset, and
+ * their respective field names can be configured using one of the provided constructors.
+ *
+ * @author Will Pruyn
+ * @author Ian Kottman
+ * @since 3.1
+ */
+public class OffsetPaginationModule extends AbstractModule {
+  private final Integer maxLimit;
```

Comment:
Is it ok to have state for a guice module class? Would it be better to have this come in through a configuration object?

Reply:
While I think it's ok to have state in the module, this has gotten a bit out of hand. I like the idea of a config object, I'll work on adding one.